actions/run-nextstrain-build-on-aws-batch/action.yaml

```diff
@@ -0,0 +1,78 @@
+name: Run Nextstrain build on AWS Batch
+description: >-
+  This GitHub Actions action is intended to be called by workflows in our other
+  repos when they need to run a nextstrain build on AWS Batch.
+
+inputs:
+  runtime-envvar-names:
+    description: >-
+      A list of environment variable names for environment variables that
+      should be passed to the build's runtime. The environment variables that
+      the Nextstrain CLI automatically forwards to the runtime do not need to be
+      listed here. See the list of environment variables that are auto-forwarded in
+      the Nextstrain CLI repo (https://github.com/nextstrain/cli/blob/master/nextstrain/cli/hostenv.py)
+
+      The environment variables can be any environment variable specified at
+      any level (workflow, job, or step) of the workflow using it.
+      The environment variable names should be separated by a space,
+      e.g. "GITHUB_RUN_ID SLACK_TOKEN SLACK_CHANNELS"
+    type: string
+    required: false
+
+  cpus:
+    description: >-
+      Number of CPUs/cores/threads/jobs to utilize at once.
+      Will be passed to the `--cpus` option for `nextstrain build` and the
+      `--cores` option for Snakemake. See `nextstrain build` docs for more details:
+      https://docs.nextstrain.org/projects/cli/page/commands/build/
+    type: number
+    default: 4
+    required: false
+
+  memory:
+    description: >-
+      Amount of memory to make available to the build. Passed to the
+      `--memory` option for `nextstrain build`.
+      See `nextstrain build` docs for more details:
+      https://docs.nextstrain.org/projects/cli/page/commands/build/
+    type: string
+    required: false
+
+  build-args:
+    description: >-
+      Additional command-line arguments to pass to `nextstrain build` after
+      the build directory (i.e. additional arguments for Snakemake).
+    type: string
+    default: ""
+    required: false
+
+runs:
+  using: composite
```

Comment:
I was going to ask why you chose a composite action over a reusable workflow, but I see that's discussed in the main page of the PR!

I would also vote for this to be a reusable workflow. I can't think of a reason why you'd need this as a step inside a job, and I imagine many caller workflows will just call this single workflow (or action) and that's it!

Reply:
I was thinking of the [adhoc configs](https://github.com/nextstrain/ncov/blob/15817aa1376eded8d02f897988de55afdd109320/.github/workflows/rebuild-gisaid.yml#L38-L48) that we create in ncov workflows for trial runs, but you're right we don't need this to be a composite workflow. 

If anything, trial run configs can be set up in a separate job and the config can be passed via outputs.